### ISSUE TITLE
TableTools Save Extension not working on shiny-server?

### ISSUE THREAD

**Issue by cdeterman** [rstudio/DT#199]
Working from the DT webpage I was trying to use the 'TableTools' extension to save a datatable.  To reproduce I use the exact same code as provided on the [DT extensions](http://rstudio.github.io/DT/extensions.html) page wrapped in a shiny app.

```r
output$tbl <- DT::renderDataTable({
  datatable(
    iris, extensions = 'TableTools', options = list(
      dom = 'T<"clear">lfrtip',
      tableTools = list(sSwfPath = copySWF())
    )
  )
})
```

This works perfectly fine if I run this in an app locally (which I then open in a browser) where I can download the iris dataset.  However, I have setup a shiny-server on a CentOS server.  When I connect to the server in my browser the table is displayed correct and I can 'copy' the data.  When I try to 'Save' a 'CSV' the popup comes up for me to choose a location for the file but it never actually saves???  No file is actually made following this.  Did I miss something that changes when DT is used with shiny-server?

**Comment by yihui** [comment]
Do you have a `www` directory in the app? You have to copy the SWF file to `www` by `copySWF('www')` as shown on the doc page http://rstudio.github.io/DT/extensions.html

**Comment by cdeterman** [comment]
yes, I added the `www` to the `copySWF` call (the `www` directory does exist) and it still won't create the file.

**Comment by cdeterman** [comment]
Could there be some server setting that would be preventing the saving perhaps?

**Comment by cdeterman** [comment]
I actually just discovered that this appears to be a Chrome issue.  I tried the same server connection with firefox and it works just fine.  I reopened with Chrome and opened the web developer tool (Ctrl-Shift-j) and noticed a 'warning'

>Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check http://xhr.spec.whatwg.org/.

This appears to have come up with TableTools [before](https://datatables.net/forums/discussion/24934/is-it-possible-to-load-the-config-of-datatables-async) but I am unsure of any solution.
